### PR TITLE
Fix race in PageCache

### PR DESCRIPTION
**Pull request by al13n321**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Just forgot to lock a mutex: https://s3.amazonaws.com/clickhouse-test-reports/60808/9a0546168094d38692725f89677077e32bd144b5/stateless_tests__tsan__[1_5]/stderr.log

### CODE REVIEW

File: src/Common/PageCache.cpp [hunk 1]
```diff
@@ -202,8 +202,11 @@ PageCache::MemoryStats PageCache::getResidentSetSize() const
     if (use_madv_free)
     {
         std::unordered_set<UInt64> cache_mmap_addrs;
-        for (const auto & m : mmaps)
-            cache_mmap_addrs.insert(reinterpret_cast<UInt64>(m.ptr));
+        {
```
Comment (Algunenano) [inline src/Common/PageCache.cpp:205]: Let's use TSA macros to protect and detect these kind of things directly in the compiler

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 4956f46562bf392a390b23bd068ae9685957b787 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60878/4956f46562bf392a390b23bd068ae9685957b787/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Common/PageCache.cpp [hunk 2]
```diff
@@ -283,6 +286,7 @@ PageCache::MemoryStats PageCache::getResidentSetSize() const
     }
 #endif
 
+    std::lock_guard lock(global_mutex);
```
Comment (Algunenano) [inline src/Common/PageCache.cpp:289]: Nice. It seems this was indeed detected by TSA

**Review by Algunenano — APPROVED** [review]

**Comment by al13n321** [comment]
Failed tests are unrelated:
 * test_system_kafka_consumers_rebalance_mv - https://github.com/ClickHouse/ClickHouse/issues/60939
 * 02784_parallel_replicas_automatic_decision_join - https://github.com/ClickHouse/ClickHouse/issues/60945
 * 02724_show_indexes - https://github.com/ClickHouse/ClickHouse/pull/60898
 * test_union_schema_inference_mode - https://github.com/ClickHouse/ClickHouse/issues/60347
 * AST fuzzer "Cannot capture column 3 because it has incompatible type: got UInt8, but LowCardinality(UInt8) is expected" - maybe https://github.com/ClickHouse/ClickHouse/pull/60451
 * performance comparison - https://github.com/ClickHouse/ClickHouse/issues/60948

**Pull request merged**